docs/ii-spec.mdx

```diff
@@ -293,34 +295,46 @@ In order to allow Internet Identity to read the path `/.well-known/ii-alternativ
 
 This section describes the interface that the backend canister provides.
 
-This interface is currently only used by its own frontend. This tight coupling means that this interface may change, even in incompatible ways. We therefore do not have to apply Candid best practices for backward-compatibility (such as using records for arguments and results).
+Note that the interface is split into 4 categories:
+* Identity management API, i.e. APIs for end-users to manage their identity
+  * Legacy API
+  * API v2 (experimental, incomplete)
+    * The aim of the API v2 is to introduce changes that cannot be made without breaking existing clients of the legacy API. While the API v2 has feature parity with the legacy API, the desired changes are not fully implemented yet:
+      * Methods should return proper results with meaningful errors
+      * Adding explicit WebAuthn signatures to security critical operations
+* HTTP gateway protocol, required to serve https://identity.ic0.app
+* Auth protocol, for creating signed delegations
+* Verifiable Credentials protocol, for creating id alias credentials
+* internal methods, not intended to be called by external clients
+    * these are methods related to initialisation of II itself and integration with its archive
 
 The summary is given by the Candid interface:
 
 <CodeBlock language="candid">{IICandidInterface}</CodeBlock>
 
-The `init_salt` method is mostly internal, see [Salt](#salt).
-
-### The `register` and `create_challenge` methods
+### Identity Management (legacy API)
```

Comment:
How about documenting the legacy in the candid interface and adding a link to it above instead of repeating the duplicating the data?

Did you discuss with someone whether we still want the endpoint descriptions in this docs?

Reply:
I did it this way because it is then aligned with how the interface spec does it. See e.g. the management canister API spec: https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-management-canister

I briefly discussed with @jessiemongeon1. But I'm open to change it and just have the candid file.
What do you think @jessiemongeon1?

Reply:
I think aligning with the Interface Spec is the best approach for consistency.